editor/src/clj/editor/material.clj

```diff
@@ -244,10 +245,11 @@
                     :type :choicebox
                     :options (protobuf-forms/make-options data-type-values)
                     :default :type-float}
-                   {:path [:element-count]
-                    :label "Count"
-                    :type :integer
-                    :default default-element-count}
+                   {:path [:vector-type]
+                    :label "Vector Type"
+                    :type :choicebox
+                    :options (protobuf-forms/make-options vector-type-values)
+                    :default default-vector-type}
```

Comment:
Q: If we have a vec4 in the shader and reconfigure it here as a mat4 - what should we do?
* Build error
* Filter out the impossible values from the dropdown?
* Both?